src/lib/docker-utils.coffee

```diff
@@ -0,0 +1,131 @@
+constants = require './constants'
+DockerToolbelt = require 'docker-toolbelt'
+{ DockerProgress } = require 'docker-progress'
+Promise = require 'bluebird'
+dockerDelta = require 'docker-delta'
+TypedError = require 'typed-error'
+_ = require 'lodash'
+{ request, resumable } = require './request'
+{ envArrayToObject } = require './conversions'
+{ checkInt } = require './validation'
+
+applyDelta = (imgSrc, deltaUrl, applyTimeout, opts, onProgress) ->
+	new Promise (resolve, reject) ->
+		req = resumable(Object.assign({ url: deltaUrl }, opts))
+		.on('progress', onProgress)
+		.on('retry', onProgress)
+		.on('error', reject)
+		.on 'response', (res) ->
+			if res.statusCode isnt 200
+				reject(new Error("Got #{res.statusCode} when requesting delta from storage."))
+			else if parseInt(res.headers['content-length']) is 0
+				reject(new Error('Invalid delta URL.'))
+			else
+				deltaStream = dockerDelta.applyDelta(imgSrc, timeout: applyTimeout)
+				res.pipe(deltaStream)
+				.on('id', (id) -> resolve('sha256:' + id))
```

Comment:
@pcarranzav I don't think you should be prepending sha here, the ID you get from docker-delta is the image ID, not the content hash, and you refer to the image with `dockerode.getImage(id)` normally.

Reply:
The image Id *is* a content hash, even if with deltas it doesn’t match the content of the image. We need to do this so that the id is the same as what you get when you do `docker.getImage(id).inspect().get(‘Id’)`

Reply:
Hmm, yes it is indeed a digest, but I thought that doing `docker.getImage('sha256:abcdef')` wouldn't do it, but it seems it works either way.